docs/bugs.json

```diff
@@ -1,4 +1,14 @@
 [
+    {
+        "uid": "SOL-2022-2",
+        "name": "NestedCallataArrayAbiReencodingSizeValidation",
+        "summary": "ABI-encoding values in calldata that contain nested dynamic arrays sometimes read beyond ``calldatasize()`` instead of reverting.",
+        "description": "Calldata validation for nested dynamic types is deferred until the first access to the nested values. Such an access may for example be a copy to memory or an index or member access to the outer type. While in most such accesses calldata validation correctly checks that the data area of the nested array is completely contained in the passed calldata (i.e. in the range [0, calldatasize()]), this check may fail, when ABI encoding such nested types again directly from calldata. For instance, this can, if a value in calldata with a nested dynamic array is passed to an external call, used in ``abi.encode`` or emitted as event. In such cases, if the data area of the nested array extends beyond ``calldatasize()``, ABI encoding it did not revert, but continue reading values from beyond ``calldatasize()`` (i.e. zero values).",
```

Comment:
Changed ``this can`` to ``this can happen`` and applied the ``continued`` check.

With regards to ``this check may *not* fail``: in my mind, we have a check that is supposed to revert in these cases, and this check failing means, it does not revert, i.e. ``this check may fail`` seems right to me... but it's indeed a bit ambiguous, so maybe I'll try to rephrase generally...

Reply:
Makes sense but the way it's written it's really a bit ambiguous and could be interpreted either way. How about clarifying it like this?

> While in most such accesses calldata validation correctly checks that the data area of the nested array is completely contained in the passed calldata (i.e. in the range [0, calldatasize()]), this check may **be performed incorrectly**, when ABI encoding such nested types again directly from calldata.